packages/components/src/components/DistributionChart.tsx

```diff
@@ -6,26 +6,41 @@ import { distributionErrorToString } from "@quri/squiggle-lang";
 import { createClassFromSpec } from "react-vega";
 import * as chartSpecification from "../vega-specs/spec-distributions.json";
 import { ErrorBox } from "./ErrorBox";
-import styled from "styled-components";
 
 let SquiggleVegaChart = createClassFromSpec({
   spec: chartSpecification as Spec,
 });
 
 type DistributionChartProps = {
   distribution: Distribution;
-  width: number;
+  width?: number;
   height: number;
 };
 
 export const DistributionChart: React.FC<DistributionChartProps> = ({
   distribution,
-  width,
   height,
+  width,
 }: DistributionChartProps) => {
+  // This code with refs and effects is a bit messy, and it's because we were
+  // having a large amount of trouble getting vega charts to be responsive. This
+  // is the solution we ended up with
+  const ref = React.useRef(null);
+  const [actualWidth, setActualWidth] = React.useState(undefined);
+
+  React.useEffect(() => {
+    // @ts-ignore
+    let getWidth = () => (ref.current ? ref.current.offsetWidth : 0);
+
+    window.addEventListener("resize", () => setActualWidth(getWidth()));
+
+    setActualWidth(getWidth());
+  }, [ref.current]);
```

Comment:
Why not ``useSize`` or another existing React hook? This feels like you're re-implementing existing functionality.

Reply:
`useSize` is not an existing react hook, it's from the `react-use` library? unless I'm missing something

Reply:
I'm not sure whether I want to add another dependency just for this

Reply:
https://github.com/streamich/react-use/blob/8ceb4c0f0c5625124f487b435a2fd0d3b3bc2a4f/src/useSize.tsx

I'm pretty nervous with this stuff. Even if you just copy + paste that one file (leaving credit in the comments), that seems safer.